design model/components.model.json

```diff
@@ -0,0 +1,551 @@
+                    "name": "Initials",
+                    "valueType": "string",
+                    "value": ""
+                },
+                {
+                    "name": "Image",
+                    "valueType": "string",
+                    "value": ""
+                },
+                {
+                    "name": "Icon",
+                    "valueType": "string"
+                }
+            ]
+        },
+        {
+            "name": "Icon",
+            "properties": [
+                {
+                    "name": "Name",
+                    "valueType": "string",
+                    "value": ""
+                },
+                {
+                    "name": "Color",
+                    "valueType": "string",
+                    "value": ""
+                }
+            ]
+        },
+        {
+            "name": "Calendar",
+            "properties": [
+                {
+                    "name": "Layout",
+                    "valueType": "enum",
+                    "enumType": {
+                        "default": "Default",
+                        "vertical": "Vertical",
+                        "horizontal": "Horizontal"
+                    },
+                    "value": "Default"
+                },
+                {
+                    "name": "ContentType",
+                    "valueType": "enum",
+                    "enumType": {
+                        "days": "Days",
+                        "months": "Months",
+                        "years": "Years"
+                    },
+                    "value": "Default"
+                },
+                {
+                    "name": "Value",
+                    "valueType": "date",
+                    "value": ""
+                },
+                {
+                    "name": "WeekStart",
+                    "valueType": "enum",
+                    "enumType": {
+                        "Sunday": "0",
+                        "Monday": "1",
+                        "Tuesday": "2",
+                        "Wednesday": "3",
+                        "Thursday": "4",
+                        "Friday": "5",
+                        "Saturday": "6"
+                    },
+                    "value": "0"
+                }
+            ]
+        },
+        {
+            "name": "Button",
+            "properties": [
+                {
+                    "name": "Type",
+                    "valueType": "enum",
+                    "enumType": {
+                        "flat": "Flat",
+                        "outlined": "Outlined",
+                        "raised": "Raised",
+                        "icon": "Icon",
+                        "fab": "Fab"
+                    }
+                },
+                {
+                    "name": "ContentType",
+                    "valueType": "enum",
+                    "enumType": {
+                        "text": "Text",
+                        "icon": "Icon"
```

Comment:
I think we should also support icon+text and text+icon

Reply:
The current sketch button does not seem to have a button+icon mode. Will such a mode be added or will the button be considered a composite component - something that can be decomposed and templated?

Reply:
I plan to add these "modes" which are actually content types/templates but am not sure in what way they will appear in the definition in order to create the sketch structure